module/skills/KickScript/src/KickScript.cpp

```diff
@@ -0,0 +1,52 @@
+#include "KickScript.hpp"
+
+#include "extension/Behaviour.hpp"
+#include "extension/Configuration.hpp"
+#include "extension/behaviour/Script.hpp"
+
+#include "message/actuation/Limbs.hpp"
+#include "message/skills/Kick.hpp"
+
+#include "utility/input/LimbID.hpp"
+
+namespace module::skills {
+
+    using extension::Configuration;
+    using extension::behaviour::Script;
+    using extension::behaviour::ScriptRequest;
+    using message::actuation::LimbsSequence;
+    using message::skills::Kick;
+    using utility::input::LimbID;
+
+    KickScript::KickScript(std::unique_ptr<NUClear::Environment> environment)
+        : BehaviourReactor(std::move(environment)) {
+
+        on<Configuration>("KickScript.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file KickScript.yaml
+            this->log_level = config["log_level"].as<NUClear::LogLevel>();
+        });
+
+        on<Provide<Kick>, Needs<LimbsSequence>>().then([this](const Kick& kick, const RunInfo& info) {
+            // If the script has finished executing, then this is Done
+            if (info.run_reason == RunInfo::RunReason::SUBTASK_DONE) {
+                emit<Task>(std::make_unique<Done>());
+                return;
+            }
+            // If it isn't a new task, don't do anything
+            if (info.run_reason != RunInfo::RunReason::NEW_TASK) {
+                emit<Task>(std::make_unique<Idle>());
+                return;
+            }
+
+            if (kick.leg == LimbID::RIGHT_LEG) {
+                emit<Script>(std::make_unique<LimbsSequence>(),
+                             std::vector<ScriptRequest>{{"Stand.yaml"}, {"KickRight.yaml"}, {"Stand.yaml"}});
```

Comment:
I would put the scripts to run in the configuration so you can change them on demand at robocup
Which you will 100% want to do

Reply:
In the past we've had to do things like swap out for a Power kick for a penalty shootout, or sometimes right kick worked but left didn't so we just set them to the same

Reply:
I think planning should take care of that